src/components/v2/index.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { Suspense, useEffect, useState } from 'react';
+import { Switch, Route } from 'react-router-dom';
+import { URLS } from '../../config';
+import { useRouteMatch, useParams, Redirect } from 'react-router';
+import { Progressing } from '../common';
+import './lib/bootstrap-grid.min.css';
+import ValuesComponent from './values/ChartValues.component';
+import AppHeaderComponent from './headers/AppHeader.component';
+import ChartHeaderComponent from './headers/ChartHeader.component';
+import { getInstalledAppDetail } from '../charts/charts.service';
+import { getInstalledChartDetail } from './appDetails/appDetails.api';
+import AppDetailsComponent from './appDetails/AppDetails.component';
+import { EnvType } from './appDetails/appDetails.type';
+import IndexStore from './appDetails/index.store';
+
+function RouterComponent({ envType }) {
+    const [isLoading, setIsLoading] = useState(true);
+    const params = useParams<{ appId: string; envId: string; nodeType: string }>();
+    const { path } = useRouteMatch();
+
+    useEffect(() => {
+        IndexStore.setEnvDetails(envType, +params.appId, +params.envId);
+
+        setIsLoading(true);
+
+        if (params.appId && params.envId) {
+            init();
+        }
+    }, [params.appId, params.envId]);
+
+    const init = async () => {
+        try {
+            let response = null;
+
+            if (envType === EnvType.CHART) {
+                response = await getInstalledChartDetail(+params.appId, +params.envId);
+            } else {
+                response = await getInstalledAppDetail(+params.appId, +params.envId);
+            }
+
+            IndexStore.setAppDetails(response.result);
```

Comment:
check if response is 200 or not, its getting stuck into loading in case of other responses for eg 404, to reproduce change envId in the url to some other value